core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -29,6 +30,16 @@ import org.apache.spark.serializer.{SerializerInstance, Serializer}
 import org.apache.spark.util.{CompletionIterator, Utils}
 
 /**
+ * Factory class that creates iterators to read records from fetched blocks
```

Comment:
To help me understand this patch's motivation a little better, it would be nice if this contained a paragraph explaining what a typical implementation of this might do (e.g. wrap a stream in a serializer) and to maybe provide one example of something that a custom implementation could do.